### PR TITLE
Add new target support for object create

### PR DESCRIPTION
**Pull request by kisbg**
JerryScript-DCO-1.0-Signed-off-by: bence gabor kis kisbg@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-object.c [hunk 1]
```diff
@@ -103,13 +104,13 @@ ecma_value_t
...
-  JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
-
   if (arguments_list_len == 0
       || ecma_is_value_undefined (arguments_list_p[0])
       || ecma_is_value_null (arguments_list_p[0]))
   {
-    return ecma_builtin_object_dispatch_construct (arguments_list_p, arguments_list_len);
+    ecma_object_t *obj_p = ecma_op_create_object_object_noarg ();
+
+    return ecma_make_object_value (obj_p);
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-object.c:113]: `return ecma_make_object_value (ecma_op_create_object_object_noarg ());`

File: jerry-core/ecma/builtin-objects/ecma-builtin-object.c [hunk 2]
```diff
@@ -124,16 +125,24 @@ ecma_value_t
...
+  if (JERRY_CONTEXT (current_new_target) != ecma_builtin_get (ECMA_BUILTIN_ID_OBJECT))
   {
-    ecma_object_t *obj_p = ecma_op_create_object_object_noarg ();
+    ecma_object_t *prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
+                                                                             ECMA_BUILTIN_ID_OBJECT_PROTOTYPE);
+    if (JERRY_UNLIKELY (prototype_obj_p == NULL))
+    {
+      return ECMA_VALUE_ERROR;
+    }
+    ecma_object_t *object_p = ecma_create_object (prototype_obj_p,
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-object.c:137]: Please add a newline before this line.

**Review by zherczeg — CHANGES_REQUESTED** [review]
Just minor things.

**Review by rerobika — APPROVED** [review]
LGTM

File: jerry-core/ecma/builtin-objects/ecma-builtin-object.c [hunk 3]
```diff
@@ -103,13 +104,12 @@ ecma_value_t
...
                                    uint32_t arguments_list_len) /**< number of arguments */
 {
-  JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
-
   if (arguments_list_len == 0
       || ecma_is_value_undefined (arguments_list_p[0])
       || ecma_is_value_null (arguments_list_p[0]))
   {
-    return ecma_builtin_object_dispatch_construct (arguments_list_p, arguments_list_len);
+
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-object.c:111]: Please don't add this newline.

**Review by zherczeg — CHANGES_REQUESTED** [review]

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**